src/fsm/TransitionsFsmBuilder.py

```diff
@@ -0,0 +1,106 @@
+import sys
+from enum import Enum
+
+from transitions import Machine
+
+from fsm.TransitionsFsmModel import TransitionsFsmModel
+from fsm.TrdFsmBuilder import TrdFsmBuilder
+
+ALL_STATES = '*'
+SAME_STATE = '='
+
+
+class TransitionsFsmBuilder(TrdFsmBuilder):
+
+    def __init__(self):
+        self.states = []
+        self.state_names = []
+        self.transitions = []
+        self.callbacks = {}
+        self.initial = None
+        self.final = None
+        self.transition_complete_callback = None
+
+    def add_transition_complete_callback(self, transition_complete_callback):
+        self.transition_complete_callback = transition_complete_callback
+
+    def add_initial_state(self, name, on_leave=None):
+        self.add_state(name, initial=True, on_leave=on_leave)
+
+    def add_final_state(self, name, on_enter=None):
+        self.add_state(name, final=True, on_enter=on_enter)
+
+    def add_state(self, state, initial=False, final=False, on_enter=None, on_leave=None, on_reenter=None):
+        if isinstance(state, Enum):
+            state = state.name
+
+        if initial:
+            self.initial = state
+        if final:
+            self.final = state
+
+        state_dict = {'name': state}
+
+        if on_enter:
+            state_dict['on_enter'] = [on_enter]
+        if on_leave:
+            state_dict['on_exit'] = [on_leave]
+        if on_reenter:
+            print("reenter not supported!", file=sys.stderr)
+
+        self.states.append(state_dict)
+        self.state_names.append(state)
+        pass
+
+    def add_global_transition(self, event, dst, on_before=None, on_after=None):
```

Comment:
add_transition requires source state, which is not applicable for global transitions which may happen from any states. The framework uses "*" notation for source state to imply any states.  I find this framework-specific and I also had trouble while using it. 

add_global_transition does not require providing source states. 

Also, IMHO global transitions will not need conditions, add_global_transition hides the condition option. In the future, this may change.